### PR TITLE
[13.0][ADD] sale_coupon_financial_risk: New module

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc @Tecnativa TT43281

This module avoids sending a coupon by mail when the costumer has exceeded it's financial risk restriction.

@chienandalu please review!

### CODE REVIEW

File: sale_coupon_financial_risk/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        exception_msg = self.evaluate_risk_message(
+            self.partner_invoice_id.commercial_partner_id
+        )
+        if exception_msg:
```
Comment (chienandalu) [inline sale_coupon_financial_risk/models/sale_order.py]: If the risk is bypassed we should send the mail (test this as well)
```suggestion
        if exception_msg and not self.env.context.get("bypass_risk"):
```

**Review by chienandalu — COMMENTED** [review]
Well done :+1: 

Just squash your commits and check pre-commit issues

**Review by chienandalu — COMMENTED** [review]
Squash both commits into a single one

**Comment by stefan-tecnativa** [comment]
> Squash both commits into a single one

Done, I was checking the pre-commit when you commented.

File: sale_coupon_financial_risk/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        exception_msg = self.evaluate_risk_message(
+            self.partner_invoice_id.commercial_partner_id
+        )
+        if exception_msg and not self.env.context.get("bypass_risk"):
+            self = self.with_context(risk_exceeded=True)
+        elif self._context.get("risk_exceeded"):
```
Comment (chienandalu) [inline sale_coupon_financial_risk/models/sale_order.py]:
```suggestion
        elif self.env.context.get("risk_exceeded"):
```

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
please review @carlosdauden

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**